dev-docs/RFCs/v4.0.0/names_for_concepts_types_functions.md

```diff
@@ -0,0 +1,131 @@
+    - for functions that can handle either hexagons or pentagons, the more general term "cell" should be used whenever possible
+- hexagon:
+    - an H3 cell which is a **topological** hexagon
+    - todo: advise use of the term "strict" for functions that can only handle hexagons?
+- pentagon:
+    - an H3 cell which is a **topological** pentagon
+- `DirectedEdge`
+    - an H3 index of `mode 2`
+    - representing a traversal from an origin cell to an adjacent destination cell
+- grid:
+    - the graph with nodes corresponding to H3 cells, and edges given by pairs of adjacent cells
+    - for example, `gridDistance` is the minimal number of edges in a graph path connecting two cells
+
+### Notes
+
+We realize that "hexagon" will still be used informally to refer to the concept of "cell" (As the development team, we do it ourselves!).
+This should be expected in casual, informal discussions of H3.
+However, when **precision** is required, we advise the use of strict technical terms like "index", "cell", "hexagon", "pentagon", etc.
+In the codebase and in the documentation, strictly correct terminology should **always** be used.
+
+
+## Use of "Topological"
+
+We should clarify in the documentation that H3 works with **topological** hexagons and pentagons, rather than true **geometric** hexagons and pentagons.
+
+See, for example: https://github.com/uber/h3-js/issues/53
+
+
+## General Function Names
+
+|          Current name         |     Proposed name     |       Notes       |
+|-------------------------------|-----------------------|-------------------|
+| *Does Not Exist (DNE)*        | `isValidIndex`        |                   |
+| `h3IsValid`                   | `isValidCell`         |                   |
+| `h3UnidirectionalEdgeIsValid` | `isValidDirectedEdge` |                   |
+| `h3IsPentagon`                | `isPentagon`          |                   |
+| `h3ToParent`                  | `cellToParent`        | or `getParent`?   |
+| `h3ToChildren`                | `cellToChildren`      | or `getChildren`? |
+| `h3IndexesAreNeighbors`       | `cellsAreNeighbors`   |                   |
+| `geoToH3`                     | `geoToCell`           |                   |
+| `h3ToGeo`                     | `cellToGeo`           |                   |
+| `numHexagons`                 | `numCells`            |                   |
+| `getRes0Indexes`              | `getRes0Cells`        |                   |
+| `getPentagonIndexes`          | `getPentagons`        |                   |
+| `h3GetBaseCell`               | `getBaseCellNumber`   |                   |
+
+todo: Do we have general guidance on when to use a `get` prefix with a function and when not?
+
+
+## H3 Grid Functions
+
+| Current name |     Proposed name      |
+|--------------|------------------------|
+| `h3Distance` | `gridDistance`         |
+| `h3Line`     | `gridPathCell`         |
+| *DNE*        | `gridPathEdge`         |
+| *DNE*        | `gridPathDirectedEdge` |
```

Comment:
I'm curious what this one would do, since the only logical value that I can think of for a valid directed edge index is `1`.

Reply:
I was thinking that `gridPathEdge` would return a collection of undirected edges connecting the passed-in origin and destination cells.

`gridPathDirectedEdge` would return a collection of directed edges connecting the origin and destination cells.

`gridPathCell` returns a collection of cells.

They're just three different ways of describing the same path between origin and destination cells, except for maybe the direction part.

Reply:
Ahh, that makes more sense. I saw it grouped with `gridDistance` and that affected my interpretation of what this would do.